### ISSUE TITLE
Must select at least one option in order for discount to be applied and not considered invalid

### ISSUE THREAD

**Issue by herbdool** [dlobo/org.civicrm.module.cividiscount#85]
I can confirm [this error](https://github.com/dlobo/org.civicrm.module.cividiscount/issues/35#issuecomment-69106674) with 4.5.4 / cividiscount 3.0. I had to individually select every price option in order for the discount to work at all. I think this is better as it's own issue so I'm referencing the comment from issue #35.

I believe this is a bug since cividiscount says:
> If your form uses a price set, the discount will be applied to all options in the set by default. However, if you want the discount to be applied only to specific price options - select those options here.

**Comment by dyguk** [comment]
We can confirm this error too. In 4.5.4/cividiscount 3.0 CiviDiscount works for event registration pages under these conditions:
1. The event doesn’t have a price set
2. The event has a price set AND you allocate the discount code to ALL the price options in the price set. It must be ALL the price options.
If these conditions aren't true, you get the message saying the code is invalid.

**Comment by adixon** [comment]
So a simple fix is just to edit that message: if your event uses a priceset, you need to include all priceset fields for which you want the discount to apply.

**Comment by dyguk** [comment]
Yes, I guess that will keep things going until there can be a fix. I imagine we'd need to add a message like, 'If any events in your selection use a price set(s), you need to include ALL the price set fields contained in all those price sets.'
The idea of being able to apply discounts to individual price set fields was rather inviting, but I guess not working for now.

**Comment by dyguk** [comment]
I've just checked this in 4.6.4 and it all seems to be working correctly now for events.
If I set a code for events with a variety of different price sets then:
If I don't set any price sets as well, the code is not applied. Correct.
If I set all price set options for all price sets for all events involved, then the code is applied for all those options. Correct.
If I set any combination (that I've tried) of price set options for any of the events then the code is correctly applied only to the price set options that I've set. Correct.

So, as far as I can see, it's all working properly now. 

Well done, team!
